pkg/controller/service/service_controller.go

```diff
@@ -164,6 +164,22 @@ func New(
 	s.serviceLister = serviceInformer.Lister()
 	s.serviceListerSynced = serviceInformer.Informer().HasSynced
 
+	nodeInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(cur interface{}) {
+				s.nodeSyncLoop()
+			},
+			UpdateFunc: func(old, cur interface{}) {
+				// We don't care about update events for nodes because we still do a full resync based on nodeSyncPeriod.
```

Comment:
Why not handle update? We should handle pacing the amount of work being done (.eg. number of times nodeSyncLoop() is called ) independent of knowing about the updates. It seems to make sense to know about the updates right now.

Reply:
I think we also need to include updates because [`nodeSyncLoop` fetches nodes using a predicate](https://github.com/kubernetes/kubernetes/blob/f6d55bcec29a1b6a2984090668e745be97a1df7e/pkg/controller/service/service_controller.go#L670) which [skips unhealthy nodes](https://github.com/kubernetes/kubernetes/blob/f6d55bcec29a1b6a2984090668e745be97a1df7e/pkg/controller/service/service_controller.go#L652-L659). My understanding is that the initial call to `AddFunc` would likely yield an unready node, so without looking at updates we wouldn't consider the node again until the node sync period became due the next time.

Overall, we wouldn't be any faster than the current implemention.

Reply:
My reasoning for leaving out `Update` was that an unhealthy node would fail LB health checks anyways and no one really relies on that status for LB traffic steering anyways. Good point on new nodes usually being `NotReady` though, will update.

Reply:
Updated but also changed resync period to 0 for node events and left the 100s period sync loop as is. Reasoning is that for every `nodeSyncPeriod` we only need to run `nodeSyncLoop` once so no need to run the predicate check per node update.